### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by AndyHale**
#### 📲 What

Updating .NET packages to their latest .NET 6 releases and third-party packages to their latest releases, for both the API and the Functional Tests projects.  Both the updates for Serilog and Open Telemetry have breaking changes for their configuration.  The appsettings.json file has been updated for Serilog's changes and startup.cs has been adjusted for Open Telemetry's changes.

In addition to the packages that have been updated, XBehave, which was used in the API project, has been deprecated so has been replaced with BDDfy.  BDDfy was chosen because, whilst it has not been updated in some time, it has not been deprecated.  It is also used in the Functional Tests project, so is a consistent choice.

#### 🤔 Why
		
Package updates have been requested for .NET and Java Stacks as part of the latest Stacks cycle.
		
#### 🛠 How
		
* Simple package updates in both csproj files.
* Breaking changes for Serilog:  The Application Insights sink's namespace had to be updated in the appsettings.json files.
* Breaking Changes for Open Telemetry: The service collection builder has been adjusted for the new fluent builder methods.

#### 👀 Evidence
		
* See csproj file changes in the 'Files Changed' tab for package updates.
* Regarding Serilog and Open Telemetry the screenshots below show that, after making the configuration changes discussed above, both Application Insights and Open Telemetry, via Jaeger, still receive logs.

open-telemetry-logs

* After the package updates, the Functional Tests project is still able to run and test the API project as expected
functional-tests-still-run
		
#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

**Review by pedroans-ensono — APPROVED** [review]

**Comment by ElvenSpellmaker** [comment]
Hey @RhysBushnell, for the Azure side of this it'll need updating to point to the new Ensono Stacks environments.

I doubt it'll work on the old stuff as we don't run pipelines for the cluster and the cert won't be valid, so Post Deploy tests will fail.

**Comment by ElvenSpellmaker** [comment]
@AndyHale Check my comment to Rhys above, we need to move it to the new cluster. Even if the build passes, the deployment is likely to fail as-is.

File: src/api/xxAMIDOxx.xxSTACKSxx.API/xxAMIDOxx.xxSTACKSxx.API.csproj [hunk 1]
```diff
@@ -8,36 +8,37 @@
     <PropertyGroup>
         <GenerateDocumentationFile>true</GenerateDocumentationFile>
         <NoWarn>$(NoWarn);1591</NoWarn>
+        <UserSecretsId>ce12d7f9-452f-4b4a-bb85-acdf9609efaf</UserSecretsId>
```
Comment (ElvenSpellmaker) [inline src/api/xxAMIDOxx.xxSTACKSxx.API/xxAMIDOxx.xxSTACKSxx.API.csproj:11]: This says secret, should it be committed?

**Pull request merged**